src/pages/EnvironmentListPage.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import {
+  Bullseye,
+  Button,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Grid,
+  GridItem,
+  Spinner,
+  Title,
+  PageSection,
+  PageSectionVariants,
+} from '@patternfly/react-core';
+import { CubesIcon } from '@patternfly/react-icons/dist/esm/icons/cubes-icon';
+import EnvironmentCard from '../components/Environment/EnvironmentCard';
+import { NamespaceContext } from '../components/NamespacedPage/NamespacedPage';
+import { EnvironmentGroupVersionKind } from '../models';
+import { EnvironmentKind } from '../types';
+import './EnvironmentListPage.scss';
+
+const EnvironmentListPage: React.FC = () => {
+  const { namespace } = React.useContext(NamespaceContext);
+  const createEnvironment = (
+    <Button variant="primary" component={(props) => <Link {...props} to="/app-studio/import" />}>
+      Create Environment
+    </Button>
+  );
+
+  const [environments, loaded] = useK8sWatchResource<EnvironmentKind[]>({
+    groupVersionKind: EnvironmentGroupVersionKind,
+    namespace,
+    isList: true,
+  });
+
+  if (!loaded) {
+    return (
+      <PageSection variant={PageSectionVariants.light} isFilled>
+        <Bullseye>
+          <Spinner />
+        </Bullseye>
+      </PageSection>
+    );
+  }
+
+  return (
+    <>
+      <PageSection variant={PageSectionVariants.light}>
+        <Title headingLevel="h4">Environments</Title>
+      </PageSection>
+      {!environments || environments.length === 0 ? (
+        <PageSection variant={PageSectionVariants.light} isFilled>
+          <Title headingLevel="h4">Environments</Title>
+          <EmptyState variant={EmptyStateVariant.large}>
+            <EmptyStateIcon icon={CubesIcon} />
+            <Title headingLevel="h4" size="lg">
+              No Environments
+            </Title>
+            <EmptyStateBody>To get started, create an environment.</EmptyStateBody>
+            {createEnvironment}
+          </EmptyState>
+        </PageSection>
+      ) : (
+        <>
+          <PageSection variant={PageSectionVariants.light}>{createEnvironment}</PageSection>
```

Comment:
This should be part of a toolbar and not page section. There should only be one page section for the whole component.